### PR TITLE
Dont check eol and accept dirs

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

File: esss_fix_format/cli.py [hunk 1]
```diff
@@ -79,8 +87,7 @@ def fix_whitespace(old_contents):
     :return unicode:
         Returns the new contents.
     """
-    input_lines = old_contents.split('\n')
-    lines = [i.rstrip('\r') for i in input_lines]
+    lines = old_contents.split('\n')
```
Comment (fogo) [inline esss_fix_format/cli.py]: Maybe would be better to use `os.linesep` instead?

Comment (gqmelo) [inline esss_fix_format/cli.py]: Actually it needs to be ```'\n'```
io.open uses ```universal newlines``` by default

But that raises another problem, when writing the new contents ```\n``` will always be converted to ```os.linesep```. But some files (e.g. *.bat) will always contain ```\r\n``` even on Linux and some (e.g. *.sh) will always contain ```\n``` even on Windows.

So we should just keep whatever line ending the file already has. This means disabling ```universal_newlines``` on ```io.open``` and using ```str.splitlines()```, which deals with any line ending. Also when joining the lines again we need to decide which line ending to use. I would say to just use the line ending of the first line

File: tests/test_esss_fix_format.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ def input_file(tmpdir):
             bravo\s\t\s
             charlie
             \tdelta
-            echo\r
+            echo
```
Comment (gqmelo) [inline tests/test_esss_fix_format.py]: Actually we should still test ```\r\n```, ```\r``` and ```\n``` line endings

Comment (nicoddemus) [inline tests/test_esss_fix_format.py]: I just pushed a change where this is no longer required: now the file is opened with universal newline support, so Python takes care of the conversion for us.

Comment (gqmelo) [inline tests/test_esss_fix_format.py]: But ```universal_newlines``` is the problem, check my other comment.

File: esss_fix_format/cli.py (hunk 1 above)
Comment (fogo) [inline esss_fix_format/cli.py]: I guess this seems like a fair and necessary change.

File: tests/test_esss_fix_format.py (hunk 2 above)
Comment (nicoddemus) [inline tests/test_esss_fix_format.py]: Done, thanks

File: esss_fix_format/cli.py [hunk 3]
```diff
@@ -15,23 +15,31 @@
...
+                    files.extend(os.path.join(root, n) for n in names if os.path.splitext(n)[1] in EXTENSIONS)
+            else:
+                files.append(file_or_dir)
     changed_files = 0
     for filename in files:
         extension = os.path.splitext(filename)[1]
         if extension in EXTENSIONS:
-            with io.open(filename, 'rb') as f:
-                old_contents = f.read().decode('UTF-8')
+            with io.open(filename, 'rU', encoding='UTF-8', newline='') as f:
```
Comment (gqmelo) [inline esss_fix_format/cli.py]: It still passing ```'rU'```

**Pull request merged**